tools/lint/main.go

```diff
@@ -0,0 +1,105 @@
+package main
+
+import (
+	"bufio"
+	_ "embed"
+	"errors"
+	"fmt"
+	"os"
+	"strings"
+)
+
+//go:embed outline.txt
+var outline string
+
+func main() {
+	if len(os.Args) < 2 {
+		fmt.Fprintln(os.Stderr, "must provide path to document to check")
+		os.Exit(1)
+	}
+
+	file, err := os.Open(os.Args[1])
+	if err != nil {
+		fmt.Fprintln(os.Stderr, "opening document:", err)
+		os.Exit(1)
+	}
+	defer file.Close()
+	scanner := bufio.NewScanner(file)
+
+	var lines []string
+	for scanner.Scan() {
+		lines = append(lines, scanner.Text())
+	}
+	if err := scanner.Err(); err != nil {
+		fmt.Fprintln(os.Stderr, "reading document:", err)
+		os.Exit(1)
+	}
+
+	anyErr := false
+
+	err = lintHeadings(lines)
+	if err != nil {
+		fmt.Fprintln(os.Stderr, err)
+		anyErr = true
+	}
+
+	err = lintEmptySections(lines)
+	if err != nil {
+		fmt.Fprintln(os.Stderr, err)
+		anyErr = true
+	}
+
+	if anyErr {
+		os.Exit(1)
+	}
+
+	fmt.Println("lint checks complete; no findings")
+}
+
+func lintHeadings(lines []string) error {
+	headings := strings.Split(outline, "\n")
```

Comment:
This has a slight bug because `outline.txt` contains an empty newline which is typical of working in vscode. You can see this for yourself with:
```
	headings := strings.Split(outline, "\n")
	fmt.Println(outline)

	for i := range headings {
		fmt.Printf("%02d: %s\n", i, headings[i])
	}
```

I found this by running the linter against the outline.txt file.
```
$ go run . outline.txt
expected heading "" from outline.txt line 270 not found
```

Reply:
Good catch, I've fixed this by running `outline` through `strings.TrimSpace`.